src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8740,10 +8740,9 @@ static bool isCandidateUnique(ref MethodSymbol? method, MethodSymbol candidate)
                 return null;
             }
 
-            if (!typeArguments.All(t => isValidTypeArgument(t.Type)))
+            if (!typeArguments.All(t => t.HasType))
             {
-                // https://github.com/dotnet/roslyn/issues/55217: Support parameter
-                // and return types that are not valid generic type arguments.
+                // Invalid parameter or return type.
                 return null;
```

Comment:
>return null; [](http://example.com/codeflow?start=16&length=12)

Is there a test that gets here? What would be an example of a scenario for this situation.

Reply:
`var d = (ref void () => { });` hits this - see `SynthesizedDelegateTypes_14()`.